netbox/dcim/views.py

```diff
@@ -346,7 +346,13 @@ def get_extra_context(self, request, instance):
             site=instance,
             position__isnull=True,
             parent_bay__isnull=True
-        ).prefetch_related('device_type__manufacturer')
+        ).prefetch_related('device_type__manufacturer').order_by('-pk')[:10]
```

Comment:
To avoid duplicating the queryset, I suggest declaring it once and then referencing that to get both the limited set (`[:10]`) and the total count. You can leave the queryset as-is here:

```python
        nonracked_devices = Device.objects.filter(
            site=instance,
            position__isnull=True,
            parent_bay__isnull=True
        ).prefetch_related('device_type__manufacturer')
```

And under the template context, do:

```
            'nonracked_devices': nonracked_devices.order_by('-pk')[:10],
            'total_nonracked_devices_count': nonracked_devices.count(),
```

Same idea with LocationView below. This helps ensure that any future changes to the queryset aren't missed in the second definition.

Reply:
Neat. I didn't realise you could do that. Updated FR pushed.